test/Microsoft.AspNet.Mvc.IntegrationTests/MutableObjectModelBinderIntegrationTest.cs

```diff
@@ -0,0 +1,1533 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.AspNet.Http;
+using Microsoft.AspNet.Http.Collections;
+using Microsoft.AspNet.Mvc.ModelBinding;
+using Xunit;
+
+namespace Microsoft.AspNet.Mvc.IntegrationTests
+{
+    // Integration tests targeting the behavior of the MutableObjectModelBinder and related classes
+    // with other model binders.
+    public class MutableObjectModelBinderIntegrationTest
+    {
+        private const string AddressBodyContent = "{ \"street\" : \"" + AddressStreetContent + "\" }";
+        private const string AddressStreetContent = "1 Microsoft Way";
+
+        private static readonly byte[] ByteArrayContent = Encoding.BigEndianUnicode.GetBytes("abcd");
+        private static readonly string ByteArrayEncoded = Convert.ToBase64String(ByteArrayContent);
+
+        private class Order1
+        {
+            public int ProductId { get; set; }
+
+            public Person1 Customer { get; set; }
+        }
+
+        private class Person1
+        {
+            public string Name { get; set; }
+
+            [FromBody]
+            public Address1 Address { get; set; }
+        }
+
+        public class Address1
+        {
+            public string Street { get; set; }
+        }
+
+        [Fact(Skip = "Extra entries in model state #2446, ModelState.Value not set due to #2445.")]
+        public async Task MutableObjectModelBinder_BindsNestedPOCO_WithBodyModelBinder_WithPrefix_Success()
+        {
+            // Arrange
+            var argumentBinder = ModelBindingTestHelper.GetArgumentBinder();
+            var parameter = new ParameterDescriptor()
+            {
+                Name = "parameter",
+                ParameterType = typeof(Order1)
+            };
+
+            // Need to have a key here so that the MutableObjectModelBinder will recurse to bind elements.
```

Comment:
I am sorry i was not clear in my comment. What I meant was if we removed `public string Name { get; set; }` then it would be created automatically. I do not see any tests with that scenario covered can we add that as well?